### PR TITLE
[Build] Use temp folder to build tests

### PR DESCRIPTION
**Pull request by cyril-sf**
Test files are build/compiled in the `build` folder and not in the `tests` folder anymore.

@Dylnuge @igillis @addepar-andy

### CODE REVIEW

File: Gruntfile.coffee [hunk 1]
```diff
@@ -255,8 +255,8 @@ module.exports = (grunt) ->
   grunt.registerMultiTask "build_test_runner_file", "Creates a test runner file.", ->
```
Comment (igillis) [inline Gruntfile.coffee]: Do we even need this task? When does it get run?

Comment (Dylnuge) [inline Gruntfile.coffee]: Jenkins uses it I think

Comment (Dylnuge) [inline Gruntfile.coffee]: @yayalice?

**Comment by Dylnuge** [comment]
LGTM but ccing @yayalice

File: Gruntfile.coffee (hunk 1 above)
Comment (cyril-sf) [inline Gruntfile.coffee]: I tried to execute it and it fails ( `tests/support/runner.html.tmpl` does not exist ). I doubt anyone is using it.

**Comment by cyril-sf** [comment]
I never merged this, but I did some extra work ( like all tests are passing now! ).

File: tests/functional/multi_select_test.coffee [hunk 2]
```diff
@@ -32,6 +31,7 @@ test "Multi select component has correct CSS classes", ->
 test "Can add item via click", ->
   expect 2
 
+  append(multiSelect)
```
Comment (azirbel) [inline tests/functional/multi_select_test.coffee]: why did you move this out of `setup`?

**Comment by azirbel** [comment]
LGTM! Hooray, it was weird to have js stuff built in the `tests` folder... and I'm very excited to have all the tests passing.
